src/components/comp_sets/fetchSelectedArticle.jsx

```diff
@@ -0,0 +1,82 @@
+import { Link, useParams } from "react-router-dom";
+import logo from "../../assets/logo.png";
+import {
+  getAllArticles,
+  getAllCommentsByID,
+  getArticleByID,
+} from "../../apis/get-api";
+import { useEffect, useState } from "react";
+import thumbsUp from "../../assets/thumbs+up.svg";
+import thumbsDown from "../../assets/thumbs+down.svg";
+import { patchArticleVote } from "../../apis/patch-api";
+
+export const FetchSelectedArticle = ({ articleVote, setArticleVote }) => {
+  let { articleid } = useParams();
+
+  const [fetchedArticle, setFetchedArticle] = useState({});
+  const [ORVote, setORVote] = useState({
+    vote: 0,
+    clicked: false,
+    upVClicked: false,
+    downVClicked: false,
+  });
+
+  useEffect(() => {
+    getArticleByID(articleid).then(({ article }) => {
+      setFetchedArticle({
+        article_id: article.article_id,
+        author: article.author,
+        body: article.body,
+        created_at: article.created_at,
+        title: article.title,
+        topic: article.topic,
+        votes: article.votes,
+      });
+    });
+  }, []);
+
+  function rateArticle(vote) {
+    setORVote((prev) => {
+      if (vote === 1 && prev.clicked === false) {
+        patchArticleVote(articleid, vote);
+        return { ...prev, vote: prev.vote + 1, clicked: !prev.clicked };
+      } else if (vote === -1 && prev.clicked === false) {
+        patchArticleVote(articleid, vote);
+        return { ...prev, vote: prev.vote - 1, clicked: !prev.clicked };
+      } else {
+        return prev;
+      }
+    });
+
```

Comment:
You could combine these conditions because you have a vote value that represents either 1 or -1. See what happens if you replace the hardcoded values of 1 in your returns with your vote parameter. 

Also to make this truly optimistic, the state change should happen before the patch, it does appear to be making that change though but you could move the patchArticleVote to after line 48 to see if it still works for you. If it does, then you don't need to repeat lines 41 and 44.